test/integration/members.test.js

```diff
@@ -185,4 +188,81 @@ describe('Members', function () {
       })
     })
   })
+
+  describe('PATCH /members/archiveMembers/:username', function () {
+    before(async function () {
+      await cleanDb()
+      const userId = await addUser(superUser)
+      jwt = authService.generateAuthToken({ userId })
+    })
+
+    it("Should return 404 if user doesn't exist", function (done) {
+      chai
+        .request(app)
+        .patch(`/members/archiveMembers/${userDoesNotExists.username}`)
+        .set('cookie', `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) { return done(err) }
+          expect(res).to.have.status(404)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.equal("User doesn't exist")
+          return done()
+        })
+    })
+
+    it('Should archive the user', function (done) {
+      addUser(userToBeArchived).then(() => {
+        chai
+          .request(app)
+          .patch(`/members/archiveMembers/${userToBeArchived.username}`)
+          .set('cookie', `${cookieName}=${jwt}`)
+          .end((err, res) => {
+            if (err) { return done(err) }
+
+            expect(res).to.have.status(204)
+            /* eslint-disable no-unused-expressions */
+            expect(res.body).to.be.a('object').to.be.empty
+
+            return done()
+          })
+      })
+    })
+
+    it('Should return 400 if user is already archived', function (done) {
+      addUser(userAlreadyArchived).then(() => {
+        chai
+          .request(app)
+          .patch(`/members/archiveMembers/${userAlreadyArchived.username}`)
+          .set('cookie', `${cookieName}=${jwt}`)
+          .end((err, res) => {
+            if (err) { return done(err) }
+
+            expect(res).to.have.status(400)
+            expect(res.body).to.be.a('object')
+            expect(res.body.message).to.equal('User is already archived')
+
+            return done()
+          })
+      })
+    })
+
+    it('Should return 401 if user is not a super_user', function (done) {
```

Comment:
Here, it should return 403 logically, right?

Reply:
Yes it should return `403` but whoever coded the middleware to check to the user has return status 401, So I have followed else we have to change it everywhere where we have used this middleware.

Reply:
@swarajpure could we do anything here with return response?

Reply:
Yes we need to make that change in the middleware itself, change should be in seperate PR

Reply:
yes, please create a ticket and tackle it in a different PR